### ISSUE TITLE
Whole YamboConvergence workchain excepts if one of the steps fails, rather than returning nicely a non-zero error code

### ISSUE THREAD

**Issue by giovannipizzi** [yambo-code/yambo-aiida#189]
I have this result:
```
YamboConvergence<6782> Excepted [1:if_(p2y_needed)(1:prepare_calculations)]
    └── YamboWorkflow<6784> Finished [300] [1:while_(can_continue)]
        ├── PwBaseWorkChain<6786> Finished [0] [7:results]
        │   └── PwCalculation<6789> Finished [0]
        └── PwBaseWorkChain<6795> Finished [300] [6:while_(should_run_process)(2:inspect_process)]
            └── PwCalculation<6798> Finished [312]
```
By looking at the report, the cause is the following:
```
[...]
2020-10-06 23:22:43 [2759 | REPORT]:   [6784|YamboWorkflow|perform_next]: last calculation failed, exiting the workflow
2020-10-06 23:22:43 [2760 | REPORT]: [6782|YamboConvergence|prepare_calculations]: setting the p2y calc as parent
2020-10-06 23:22:43 [2761 | REPORT]: [6782|YamboConvergence|on_except]: Traceback (most recent call last):
  File "/home/pizzi/git/aiida-core/aiida/orm/utils/managers.py", line 83, in __getattr__
    return self._get_node_by_link_label(label=name)
  File "/home/pizzi/git/aiida-core/aiida/orm/utils/managers.py", line 64, in _get_node_by_link_label
    return self._node.get_outgoing(link_type=self._link_type).get_node_by_label(label)
  File "/home/pizzi/git/aiida-core/aiida/orm/utils/links.py", line 302, in get_node_by_label
    raise exceptions.NotExistent('no neighbor with the label {} found'.format(label))
aiida.common.exceptions.NotExistent: no neighbor with the label remote_folder found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pizzi/.virtualenvs/aiida-dev/lib/python3.7/site-packages/plumpy/process_states.py", line 225, in execute
    result = self.run_fn(*self.args, **self.kwargs)
  File "/home/pizzi/git/aiida-core/aiida/engine/processes/workchains/workchain.py", line 205, in _do_step
    finished, stepper_result = self._stepper.step()
  File "/home/pizzi/.virtualenvs/aiida-dev/lib/python3.7/site-packages/plumpy/workchains.py", line 268, in step
    finished, result = self._child_stepper.step()
  File "/home/pizzi/.virtualenvs/aiida-dev/lib/python3.7/site-packages/plumpy/workchains.py", line 400, in step
    finished, retval = self._child_stepper.step()
  File "/home/pizzi/.virtualenvs/aiida-dev/lib/python3.7/site-packages/plumpy/workchains.py", line 268, in step
    finished, result = self._child_stepper.step()
  File "/home/pizzi/.virtualenvs/aiida-dev/lib/python3.7/site-packages/plumpy/workchains.py", line 219, in step
    return True, self._fn(self._workchain)
  File "/home/pizzi/git/plugins_aiida/yambo-aiida/aiida_yambo/workflows/yamboconvergence.py", line 248, in prepare_calculations
    set_parent(self.ctx.calc_inputs, self.ctx.p2y.outputs.remote_folder)
  File "/home/pizzi/git/aiida-core/aiida/orm/utils/managers.py", line 89, in __getattr__
    "Node '{}' does not have an input with link '{}'".format(self._node.pk, name)
aiida.common.exceptions.NotExistentAttributeError: Node '6784' does not have an input with link 'remote_folder'
```
That indeed failed as follows:
```
$ verdi process show 6784
Property     Value
-----------  ----------------------------------------------------------------
type         YamboWorkflow
state        Finished [300] The workchain failed with an unrecoverable error.
pk           6784
[...]
```
with a report:
```
2020-10-06 23:07:04 [2740 | REPORT]: [6784|YamboWorkflow|start_workflow]: no previous pw calculation found, we will start from scratch
2020-10-06 23:07:04 [2741 | REPORT]: [6784|YamboWorkflow|start_workflow]:  workflow initilization step completed.
2020-10-06 23:07:05 [2742 | REPORT]: [6784|YamboWorkflow|can_continue]: the workflow continues with a scf calculation
2020-10-06 23:07:05 [2743 | REPORT]: [6784|YamboWorkflow|perform_next]: performing a scf calculation
2020-10-06 23:07:06 [2744 | REPORT]:   [6786|PwBaseWorkChain|run_process]: launching PwCalculation<6789> iteration #1
2020-10-06 23:15:58 [2745 | REPORT]:   [6786|PwBaseWorkChain|results]: work chain completed after 1 iterations
2020-10-06 23:15:58 [2746 | REPORT]:   [6786|PwBaseWorkChain|on_terminated]: remote folders will not be cleaned
2020-10-06 23:15:58 [2747 | REPORT]: [6784|YamboWorkflow|can_continue]: the workflow continues with a nscf calculation
2020-10-06 23:15:58 [2748 | REPORT]: [6784|YamboWorkflow|perform_next]: performing a nscf calculation
2020-10-06 23:15:59 [2749 | REPORT]:   [6795|PwBaseWorkChain|run_process]: launching PwCalculation<6798> iteration #1
2020-10-06 23:22:42 [2753 | REPORT]:   [6795|PwBaseWorkChain|report_error_handled]: PwCalculation<6798> failed with exit status 312: The stdout output file was incomplete probably because the calculation got interrupted.
2020-10-06 23:22:42 [2754 | REPORT]:   [6795|PwBaseWorkChain|report_error_handled]: Action taken: unrecoverable error, aborting...
2020-10-06 23:22:42 [2755 | REPORT]:   [6795|PwBaseWorkChain|inspect_process]: PwCalculation<6798> failed but a handler detected an unrecoverable problem, aborting
2020-10-06 23:22:42 [2756 | REPORT]:   [6795|PwBaseWorkChain|on_terminated]: remote folders will not be cleaned
2020-10-06 23:22:42 [2757 | REPORT]: [6784|YamboWorkflow|can_continue]: the workflow continues with a yambo calculation
2020-10-06 23:22:42 [2758 | REPORT]: [6784|YamboWorkflow|perform_next]: performing a yambo calculation
2020-10-06 23:22:43 [2759 | REPORT]: [6784|YamboWorkflow|perform_next]: last calculation failed, exiting the workflow
```
Finally, the PwCalculation failed with the state `Finished [312] The stdout output file was incomplete probably because the calculation got interrupted.` and indeed pw crashes with:
```
$ verdi calcjob outputcat  6798 | tail

 Really copied et H->D
 Really copied vrs H->D
 zhegvdx_gpu error: cusolverDnZpotrf failed!

 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     Error in routine  cdiaghg_gpu (1):
      zhegvdx_gpu failed
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

     stopping ...
```
(memory error? Not sure)

Anyway, I think this error should be caught nicely by the top workchain, that should detect that something went wrong in the called YamboWorkflow and stop with an appropriate error code, rather than excepting.

**Comment by mikibonacci** [comment]
Hi Giovanni, you're right, I'm working on it. 
The PwCalculation is indeed a memory error, I think it needs more GPUs.

**Comment by mikibonacci** [comment]
Solved in release v1.1.3
